### DISCUSSION TITLE
Example project of stripe payments with firebase using Realtime Database?

### DISCUSSION THREAD

**Discussion by makinggainz** [firebase/functions-samples#840] [Q&A]
Hello, I am looking for an example project [like this one](https://github.com/firebase/functions-samples/tree/master/stripe) but using the Realtime Databse, instead of Firestore. 

More specifically (If its possible): 

- I'm looking for a project which makes stripe subscriptions payments work using firebase as the backend, and iOS on the front end.

**Comment by ryanmio** [comment]
### Suggested Approach:

1. **Database**: Use Firebase Realtime Database instead of Firestore. It has similar features like querying and real-time updates but with a different API.

2. **Stripe API**: Use Stripe's Node.js SDK on a Cloud Function to handle payments.

3. **iOS Frontend**: Use Stripe's iOS SDK to gather card information securely.

### Sample Cloud Function for Stripe:

```javascript
const functions = require('firebase-functions');
const admin = require('firebase-admin');
const stripe = require('stripe')(YOUR_STRIPE_SECRET_KEY);

admin.initializeApp();

exports.createStripeCharge = functions.https.onCall(async (data, context) => {
  // Authentication and data validation here
  
  const amount = data.amount;
  const currency = data.currency;
  
  // Create Stripe Charge
  const charge = await stripe.charges.create({
    amount,
    currency,
    // ... other charge details
  });
  
  // Record the charge in the Realtime Database
  const db = admin.database();
  const ref = db.ref('payments');
  return ref.push(charge);
});
```

In iOS:
Use Stripe's iOS SDK to send the payment token to the Cloud Function for actual charge.


This is just a high-level example. Let me know if you need further elaboration.
